packages/gatsby-theme-learning/src/hooks/use-course-status.ts

```diff
@@ -0,0 +1,59 @@
+import useSWR from 'swr';
+import { useContext, useEffect, useState } from 'react';
+import ConfigContext from '../components/config-context';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useAuthToken } from './use-auth-token';
+import type { Course } from '../external-types';
+
+const fetcher = async (url: string, accessToken: string) => {
+  const data = await fetch(url, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      Authorization: `Bearer ${accessToken}`,
+    },
+  });
+  return data.json();
+};
+
+const useFetchCourses = () => {
+  const [authToken, setAuthToken] = useState<string | undefined>();
+  const { isAuthenticated } = useAuth0();
+  const { learnApiBaseUrl } = useContext(ConfigContext);
+  const { getAuthToken } = useAuthToken();
+  const apiEndpoint = `${learnApiBaseUrl}/api/courses`;
+  useEffect(() => {
+    const fetchToken = async () => {
+      const token = await getAuthToken();
+      setAuthToken(token);
+    };
+    fetchToken();
+  });
+
+  const { data, error, isLoading } = useSWR(
+    isAuthenticated && authToken ? apiEndpoint : null,
```

Comment:
As you can see here we're using conditional fetching. Basically it will fetch the courses only when authenticated and with a valid authToken (probably we just need to check the authToken). The issue here is that all the renderings that happen while the authToken has been fetched, swr does not return the stale data but instead:
```
{
data: undefined
isLoading: false
error: undefined
```
This is a probelm because it causes a very quick flash in the rendering each time we navigate through pages.
I spent quite some time trying to figure out how to overcome this issue without having to use the imperative API for fetching the data. Maybe I'm missing the big picture here, any ideas?

Reply:
Ignore this comment. After reading a bit more in depth the documentation, I refactored the code and now it seems to be working as expected